Feature/easy bootstrap

matejart opened the pull request:
This set of changes:

* Updates the installation instructions - the part that involves deploying the Cloudify blueprint. The process should now be clearer, easier to follow and with less time consuming steps.
* Adds helper scripts for simpler installation of the DMon. Part of the reason for this is to make installation of just DMon simpler. Another reason is to make this process an integral step for installing [Cloudify Manager, DICE Deplopyment Service and DMon](https://github.com/dice-project/DICE-Deployment-Service/blob/master/doc/AdminGuide.md)
* Bring the upstart jobs for `dmon-es` and `dmon-ls`, replacing the System V scripts with all essential features (except for the log files, which might get stuck in `/var/log/upstart`).
* Updates the DMon code to use the upstart parameter passing syntax.
* Brings the blueprint up-to-date with the unified infrastructe plug-in definition.

matejart committed:
Update Cloudify bootstrap scripts

Extended the blueprint and the scripts to extract the outputs
from the DMon blueprint deployment to gather the data needed for
forming inputs for DDS configuration.

matejart committed:
Update bootstrap documentation

Added a section about updating configuration of the DICE Deployment
Service, so that the deployments will be able to use the DMon for
registering applications. The text to focus on those who have
Cloudify already set up, so it is more straightforward.

matejart committed:
Fix handling of dmon-ls service

Changed the way the parameters are assigned to an upstart job. Also
removed all the PID handling, because upstart does it all for us.
The restart action is indepotent, so it works both as start and
restart. The PID is then properly handled internally, therefore
we can set the value in the DMon to -1.

matejart committed:
Replace System V scripts with Upstart scripts

Changed `dmon-es` and `dmon-ls` to Upstart for compatibility
with the dmon code.

matejart committed:
Update elasticsearch (re)starting code for upstart

Updated the POST to `/dmon/v2/overlord/core/ls` to work with
upstart. This changes the behaviour by no longer needing to manage
the PIDs. It also fixes the supplying of command line parameter
for heap size.

matejart committed:
Update the blueprint lib import URL to 0.7.1

This change ensures that we use the latest Chef blueprints for
installing DICE Monitoring Framework.

tadeboro commented on bootstrap.sh:
```diff
@@ -108,14 +106,12 @@ tar zxf logstash-2.2.1.tar.gz
 echo "Setting up logrotate ..."
 
-echo "/opt/IeAT-DICE-Repository/src/logs/logstash.log{
+echo "/var/log/upstart/dmon-ls.log{
```
Logstash logs are stored in /opt/DICE-Monitoring/src/logs/logstash.log (upstart script passes "-l /opt/DICE-Monitoring/src/logs/logstash.log" flag to logstash agent command), upstart logs only contain standard output and error from logstash command, which should be mostly empty.

matejart replied:
Thanks for spotting this. I was thinking about elasticsearch, which does not have the redirect of log.

tadeboro commented on src/pyDMON.py:
```diff
@@ -4056,85 +3971,24 @@ def post(self):
-            app.logger.info("[%s] : [INFO] LS Core started with PID %s, Storm %s and YanrHistory %s",
-                                 datetime.fromtimestamp(time.time()).strftime('%Y-%m-%d %H:%M:%S'), lsPID, stormStatus, yarnStatus)
+        try:
+            subprocess.check_call(["service", "dmon-ls", "restart",
```
Same argument as with dmon-es service restart.

tadeboro commented on src/pyDMON.py:
```diff
@@ -2760,45 +2698,22 @@ def post(self, hostFQDN):
-            response = jsonify({'Status': 'ES Core Started', 'PID': esPID})
-            response.status_code = 201
+        try:
+            subprocess.check_call(["service", "dmon-es", "restart",
```
Maybe it would be safer to call "initctl restart dmon-es ..." directly here in order to avoid service wrapper? I'm a bit worried that service wrapper might not pass all of the parameters to the upstart and things could get messy and hard to debug.

tadeboro reviewed (changes requested):
Changes in this pull request are generally OK, but before things get merged, logstash log path needs to be fixed in logrotate config.

Another thing that is bothering me is the fact that explicit kill command has not been removed from src/pyDMON.py. I would comment on that inline, but the Github review system does not allow me to do so, so I brought this up in here.

matejart committed:
Switch from service wrapper to initctl

To be safer, we now use `initctl` directly instead of calling
a service wrapper.

matejart commented:
I have addressed all these comments except for the kill commands.
